src/sql/workbench/contrib/connection/browser/connectionStatus.ts

```diff
@@ -58,17 +62,52 @@ export class ConnectionStatusbarItem extends Disposable implements IWorkbenchCon
 
 	// Update the connection status shown in the bar
 	private _updateStatus(): void {
-		let activeConnection = TaskUtilities.getCurrentGlobalConnection(this._objectExplorerService, this._connectionManagementService, this._editorService, this._logService);
+		let activeConnection: IConnectionProfile = TaskUtilities.getCurrentGlobalConnection(this._objectExplorerService, this._connectionManagementService, this._editorService, this._logService);
+		let connectionId: string = undefined;
+		let editorNewInfo: ConnectionManagementInfo = this.getCurrentActiveEditorInfo();
+		// retrieve the currently focused editor so we can get the correct connection id value for it.
+		if (editorNewInfo) {
+			// We only want to display the connection ID for editor connections, so first check to make
+			// sure the connection we're displaying information for (which may be from OE or other
+			// sources) matches the current active editor before updating the ID
+			if (editorNewInfo && editorNewInfo?.connectionProfile?.id === activeConnection?.id) {
+				connectionId = editorNewInfo?.serverConnectionId;
+			}
+		}
 		if (activeConnection) {
-			this._setConnectionText(activeConnection);
+			this._setConnectionText(activeConnection, connectionId)
 			this.show();
-		} else {
+		}
+		else {
 			this.hide();
 		}
 	}
 
+	// If the connection server id for URI changes, we need to update the info for the URI with the new one
+	// If the current editor is the one being updated, we will do a refresh if the id is different.
+	private _updateUriForInfo(uriToUpdate: string, idForUpdate: string) {
+		let newInfo: ConnectionManagementInfo = this._connectionManagementService.getConnectionInfo(uriToUpdate);
+		let isDifferent: boolean = false;
+		if (newInfo && newInfo.serverConnectionId !== idForUpdate) {
+			isDifferent = true;
+			newInfo.serverConnectionId = idForUpdate;
```

Comment:
please find a better name for the method. also we should use /**/  format for method comments.